In [2]:
import pandas as pd

In [29]:
df = pd.read_excel("movies.xlsx")
df

,MOVIES,YEAR,GENRE,RATING,ONE-LINE,STARS,VOTES,RunTime,Gross
0,Pokémon: Happy Birthday to You!,(2017 TV Special),"\nAnimation, Short",5.4,"\nToday is your big day, and Ash, Pikachu and ...",\n,21000.0,10.0,NaN
1,Beat Bugs: Happy Birthday to You!,(2017 TV Special),"\nAnimation, Short",0.0,\nThey say it's your birthday - and there's no...,\n,0.0,10.0,NaN
2,Project Mc²: Happy Birthday to You!,(2017 TV Special),\nShort,0.0,\nMcKeyla and the agents of NOV8 have chosen y...,\n,0.0,30.0,NaN
3,The Sense8 Experience,(2017),\nShort,8.9,\nAdd a Plot\n,\n Director:\nJosh Oreck\n| \n Stars:\nA...,736000.0,40.0,NaN
4,Night of 1000 Meryls: Drag Queens Perform Mery...,(2020 Video),"\nShort, Comedy, Music",0.0,\nAdd a Plot\n,\n Director:\nThom Kerr\n| \n Stars:\nMa...,0.0,50.0,NaN
...,...,...,...,...,...,...,...,...,...
995,The 39 Steps,NaN,"\nDrama, Thriller",0.0,"\nFollows an ordinary man, who becomes an unwi...",\n \n Star:\nBenedict Cumberbatch\n,0.0,NaN,NaN
996,On the Verge,(2021– ),\nComedy,0.0,\nAdd a Plot\n,\n Director:\nJulie Delpy\n| \n Stars:\n...,0.0,NaN,NaN
997,Toy Boy,(2019),"\nCrime, Drama, Thriller",7.5,\nWith Hugo once again in prison for Philip's ...,\n Director:\nJavier Quintas\n| \n Stars...,119000.0,NaN,NaN
998,Connected,NaN,\nDocumentary,0.0,\nExploring the intricate and surprising conne...,\n \n Star:\nLatif Nasser\n,0.0,NaN,NaN


# 5


In [11]:
df["STARS"].duplicated().sum()

np.int64(175)

In [12]:
NF = df.drop_duplicates("STARS")
NF

,MOVIES,YEAR,GENRE,RATING,ONE-LINE,STARS,VOTES,RunTime,Gross
0,Pokémon: Happy Birthday to You!,(2017 TV Special),"\nAnimation, Short",5.4,"\nToday is your big day, and Ash, Pikachu and ...",\n,21000.0,10.0,NaN
3,The Sense8 Experience,(2017),\nShort,8.9,\nAdd a Plot\n,\n Director:\nJosh Oreck\n| \n Stars:\nA...,736000.0,40.0,NaN
4,Night of 1000 Meryls: Drag Queens Perform Mery...,(2020 Video),"\nShort, Comedy, Music",0.0,\nAdd a Plot\n,\n Director:\nThom Kerr\n| \n Stars:\nMa...,0.0,50.0,NaN
7,Puffin Rock,(2015–2016),"\nAnimation, Family",8.7,"\nPuffin Rock follows the adventures of Oona, ...","\n \n Stars:\nKate McCafferty, \...",783000.0,70.0,NaN
8,Hey Duggee,(2014– ),"\nAnimation, Family",8.9,"\nDuggee looks after the Squirrels, who earn d...",\n \n Stars:\nAlexander Armstron...,881000.0,70.0,NaN
...,...,...,...,...,...,...,...,...,...
993,Fast & Furious Spy Racers,(2019– ),"\nAnimation, Action, Crime",7.8,\nMs. Nowhere tells the crew that the mission ...,\n Director:\nJames Yang\n| \n Stars:\nT...,69000.0,NaN,NaN
996,On the Verge,(2021– ),\nComedy,0.0,\nAdd a Plot\n,\n Director:\nJulie Delpy\n| \n Stars:\n...,0.0,NaN,NaN
997,Toy Boy,(2019),"\nCrime, Drama, Thriller",7.5,\nWith Hugo once again in prison for Philip's ...,\n Director:\nJavier Quintas\n| \n Stars...,119000.0,NaN,NaN
998,Connected,NaN,\nDocumentary,0.0,\nExploring the intricate and surprising conne...,\n \n Star:\nLatif Nasser\n,0.0,NaN,NaN


Encontramos que la variable stars tiene 175 duplicados, donde el mejor tratamiento es dejarlo como esta por que estos datos tienen la posibilidad de repetirse ya que el identificador es la pelicula por lo tanto el director y las estrellas pueden estar varias veces. Por otro lado al eliminar los duplicados quedan 825 observaciones por las mismas 9 columnas.

# 6 

Teniendo en cuenta que es el año y rango de años de emision empezaria por reemplazar los caracteres especiales por un guion para una mayor facilidad a la hora de leer la base de datos.Ademas con los rangos abiertos entenderia si es un error en la captura de datos o si se debe a que todavia la estan emitiendo a dia de hoy para asi tener claridad de lo que me dicen los datos.

# 7 

Que el rating sea 0 puede indicar que falta el dato ya que el promedio se mueve de 1.1 a 9.9 por lo cual ese 0 indicaria que es un valor faltante, por lo cual agregarlo al promedio de rating en general generaria un sesgo hacia abajo teniendo como promedio un valor menor al que deberia ser, lo mismo pasaria si hubieran valores mayores a 9.9 ya que indicarian un mala toma de los datos y generaria un sesgo hacia arriba de lo que realmente es. Por lo tanto debe tratarse el 0 como un valor faltante y no debe hacer parte de las evaluaciones a menos de que negocio lo pida de tal manera, sin embargo es importante conocer por que hace falta este dato y conocer si se puede obtener la informacion correspondiente.

In [20]:
df["RATING"].unique()


array([5.4, 0. , 8.9, 8.7, 6.5, 6.2, 5.2, 8.2, 7.2, 4.9, 8.5, 7.5, 6.4,
       7.7, 8. , 7.6, 8.3, 8.8, 8.1, 6.9, 7. , 8.4, 6.1, 6.7, 7.9, 6.6,
       5.8, 9. , 9.2, 7.3, 6.8, 4.8, 9.7, 9.1, 7.1, 6.3, 9.3, 5.6, 5. ,
       7.4, 7.8, 6. , 3.7, 8.6, 5.5, 3.8, 5.1, 5.3, 9.8, 5.9, 2.5, 5.7,
       9.5, 3.6, 4.3, 3.1, 4.6, 3.4, 3.2, 4.5, 4.7, 4.4, 1.8, 2.7, 4.1,
       3.3, 2.9, 4.2, 2.3, 2.8, 4. ])

# 8 

Al tomar decisiones en una sola metrica se dejan por fuera en este caso valores como la cantidad de personas que la vieron generando sesgos en el resultado por lo tanto se debe tener en cuenta si existen relaciones entre las varibles de manera logica y numerica.
En este caso yo tendria en cuenta las variables "RATING" y "VOTES" de manera que los que tengan mas votos y mejor rating queden mas arriba ya que indica una mejor aprobacion por parte de un gran publico, yo tendria en cuenta los valores donde los votos sean mayor a la media ya que representarian un grupo grande en general de personas.